app/models/vet360_redis/contact_information.rb

```diff
@@ -0,0 +1,166 @@
+    # @return [Vet360::Models::Email] The user's one email address model
+    #
+    def email
+      return unless @user.loa3?
+
+      value_for('emails')&.first
+    end
+
+    # Returns the user's residence. In Vet360, a user can only have one
+    # residence address.
+    #
+    # @return [Vet360::Models::Address] The user's one residence address model
+    #
+    def address
+      return unless @user.loa3?
+
+      dig_out('addresses', 'address_pou', Vet360::Models::Address::RESIDENCE)
+    end
+
+    # Returns the user's mailing address. In Vet360, a user can only have one
+    # mailing address.
+    #
+    # @return [Vet360::Models::Address] The user's one mailing address model
+    #
+    def mailing_address
+      return unless @user.loa3?
+
+      dig_out('addresses', 'address_pou', Vet360::Models::Address::CORRESPONDENCE)
+    end
+
+    # Returns the user's home phone. In Vet360, a user can only have one
+    # home phone.
+    #
+    # @return [Vet360::Models::Telephone] The user's one home phone model
+    #
+    def home_phone
+      return unless @user.loa3?
+
+      dig_out('telephones', 'phone_type', Vet360::Models::Telephone::HOME)
+    end
+
+    # Returns the user's mobile phone. In Vet360, a user can only have one
+    # mobile phone.
+    #
+    # @return [Vet360::Models::Telephone] The user's one mobile phone model
+    #
+    def mobile_phone
+      return unless @user.loa3?
+
+      dig_out('telephones', 'phone_type', Vet360::Models::Telephone::MOBILE)
+    end
+
+    # Returns the user's work phone. In Vet360, a user can only have one
+    # work phone.
+    #
+    # @return [Vet360::Models::Telephone] The user's one work phone model
+    #
+    def work_phone
+      return unless @user.loa3?
+
+      dig_out('telephones', 'phone_type', Vet360::Models::Telephone::WORK)
+    end
+
+    # Returns the user's temporary phone. In Vet360, a user can only have one
+    # temporary phone.
+    #
+    # @return [Vet360::Models::Telephone] The user's one temporary phone model
+    #
+    def temporary_phone
+      return unless @user.loa3?
+
+      dig_out('telephones', 'phone_type', Vet360::Models::Telephone::TEMPORARY)
+    end
+
+    # Returns the user's fax number. In Vet360, a user can only have one
+    # fax number.
+    #
+    # @return [Vet360::Models::Telephone] The user's one fax number model
+    #
+    def fax
+      return unless @user.loa3?
+
+      dig_out('telephones', 'phone_type', Vet360::Models::Telephone::FAX)
+    end
+
+    # The status of the last Vet360::ContactInformation::Service response,
+    # or not authorized for for users < LOA 3
+    #
+    # @return [Integer <> String] the status of the last Vet360::ContactInformation::Service response
+    #
+    def status
+      return Vet360::ContactInformation::PersonResponse::RESPONSE_STATUS[:not_authorized] unless @user.loa3?
+
+      response.status
+    end
+
+    # @return [Vet360::ContactInformation::PersonResponse] the response returned from
```

Comment:
I would add a comment saying the _potentially cached_ response from Vet360

Reply:
👍